checker/tests/initialization/TestPolyInitialized.java

```diff
@@ -0,0 +1,38 @@
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.initialization.qual.NotOnlyInitialized;
+import org.checkerframework.checker.initialization.qual.PolyInitialized;
+import org.checkerframework.checker.initialization.qual.UnknownInitialization;
+
+public class TestPolyInitialized {
+
+    @NotOnlyInitialized String testStr;
+
+    String test = "test";
+
+    TestPolyInitialized(@UnknownInitialization String str) {
+        this.testStr = identity(str);
+        // :: error: (assignment.type.incompatible)
+        this.test = identity(str);
+    }
+
+    @PolyInitialized String identity(@UnknownInitialization TestPolyInitialized this, @PolyInitialized String str) {
+        return str;
+    }
+
+    void test1() {
+        @UnknownInitialization String receiver = identity(testStr);
+    }
+
+    void test2() {
+        @Initialized String receiver = identity(test);
+    }
+
+    @Initialized String test3(@UnknownInitialization String str) {
+        // :: error: (return.type.incompatible)
+        return identity(str);
+    }
+
```

Comment:
Can you think of a test case that makes sure the type frames are correctly substituted? That is, call `identity` on something that takes a `@UnknownInitialization(SomeFrame.class)` and then assign the result to a variable of the same type.

Reply:
Ok, I will put it in the same method. I am wondering why it is called a type frame.

Reply:
I don't see a test with a type frame.
See https://eisop.github.io/cf/manual/manual.html#initialization-checker and in particular the "Freedom before Commitment" paper.

Reply:
improve the link - aosen